openwisp_firmware_upgrader/base/models.py

```diff
@@ -761,6 +804,9 @@ def upgrade(self, recoverable=True):
         # meaning the image file is corrupted or not apt for flashing
         except UpgradeAborted:
             self.status = "aborted"
+        # this exception is raised when the upgrade is cancelled by the user
+        except UpgradeCanceled:
+            self.status = "cancelled"
```

Comment:
I think this is redundant, because the `cancel` method is already flagging the operation as `cancelled`.

So here, passing should be enough:

```suggestion
            pass
```